### ISSUE TITLE
http: abort when closing with pipelining requests in flight

### ISSUE THREAD

**Issue by mcollina** [nodejs/undici#753]
* **Version**: v16.0.0
* **Platform**: linux
* **Subsystem**: http

### What steps will reproduce the bug?

Checkout https://github.com/fastify/fastify and run `npm i`:

```
$ node test/close-pipelining.test.js
TAP version 13
# Subtest: Should return 503 while closing - pipelining
    ok 1 - should not error
node[1156327]: ../src/node_http_parser.cc:567:static void node::{anonymous}::Parser::Initialize(const v8::FunctionCallbackInfo<v8::Value>&): Assertion `args[3]->IsInt32()' failed.
 1: 0xb12b00 node::Abort() [node]
 2: 0xb12b7e  [node]
 3: 0xb2a282  [node]
 4: 0xd5f70b  [node]
 5: 0xd60bac  [node]
 6: 0xd61226 v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [node]
 7: 0x160c579  [node]
Aborted
```

I will work on a closed reproduction soon.

### How often does it reproduce? Is there a required condition?

Always

### What is the expected behavior?

not abort

**Comment by mcollina** [comment]
transfered from Node.js

**Comment by indutny** [comment]
Why would it be unidici? I thought you were using wasm build of llhttp and not the bindings that Node provides. The crash seems to happen in `src/node_http_parser.cc`. Also, I can't reproduce it.

**Comment by indutny** [comment]
Ah, `npm test` fails indeed.

**Comment by mcollina** [comment]
I fixed it, it's the previous version.

**Comment by mcollina** [comment]
thanks for the help!

**Comment by indutny** [comment]
Ah, awesome. Glad it worked!

**Comment by mcollina** [comment]
Fixed in Undici v3.3.5.
